internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -195,16 +195,24 @@ func (r *ClusterDeploymentsReconciler) Reconcile(origCtx context.Context, req ct
 		return r.updateStatus(ctx, log, clusterInstall, cluster, err)
 	}
 
-	if clusterDeployment.Spec.PullSecretRef == nil {
-		errorMessage := "Missing reference to pull secret"
+	if clusterDeployment.Spec.PullSecretRef == nil && !clusterDeployment.Spec.Installed {
+		errorMessage := "missing reference to pull secret"
 		log.WithError(err).Error(errorMessage)
 		return r.updateStatus(ctx, log, clusterInstall, cluster, newInputError(errorMessage))
 	}
 
 	// Make sure that the PullSecret Secret has the needed label
-	_, err = getSecret(ctx, r.Client, r.APIReader, types.NamespacedName{Namespace: req.Namespace, Name: clusterDeployment.Spec.PullSecretRef.Name})
-	if err != nil {
-		log.WithError(err).Error("error setting label on Pull Secret")
+	if clusterDeployment.Spec.PullSecretRef != nil {
+		_, err = getSecret(ctx, r.Client, r.APIReader, types.NamespacedName{Namespace: req.Namespace, Name: clusterDeployment.Spec.PullSecretRef.Name})
+		if err != nil {
+			log.WithError(err).Error("error setting label on Pull Secret")
+			return r.updateStatus(ctx, log, clusterInstall, cluster, err)
+		}
+	}
+
+	// Make sure that the ImageSetRef is set for clusters not already installed
+	if clusterInstall.Spec.ImageSetRef == nil && !clusterDeployment.Spec.Installed {
+		log.WithError(err).Error("missing ImageSetRef for cluster that is not installed")
```

Comment:
What value will have `err` here? I think you need to create an error.

Reply:
Yep